### PR TITLE
Updates testing map with a tram and adds tram collision behavior

### PR DESCRIPTION
**Pull request by Vivalas2**
Up for review. The tram code seems laggy in general, so optimizations are needed. I fixed a lot of issues with gibbing creating lag because the game was trying to gib the same thing multiple times, using a list, and implements basic collision behavior.

### CODE REVIEW

**Comment by Vivalas2** [comment]
image


Supplementary video

File: code/modules/tram/tram.dm [hunk 1]
```diff
@@ -229,8 +247,38 @@
...
+			crash_list -= A
 	return 1
 
+
+obj/tram/tram_controller/proc/handle_crash(var/atom/movable/A,var/dir)
+	if(!istype(A)) return
+	if((A in gibbed)) return //so we don't gib things multiple times
+	if(istype(A,/mob/living))
+		if(istype(A,/mob/living/carbon))
+			if(A:lying)
```
Comment (GinjaNinja32) [inline code/modules/tram/tram.dm]: please don't use `:`

File: code/modules/tram/tram.dm [hunk 2]
```diff
@@ -229,8 +247,38 @@
...
+				A:gib()
+				gibbed += A
+			else //if it's not lying down, push it and break its face
+				A:apply_damage(5) //brute by default, 50 damage in one second
+				if(prob(2)) A:Weaken(2) //RNGesus is not in your favor, RIP buddy...
+				step_towards(A, get_step(get_turf(A),dir))
+				crash_list -= A
+		else
+			crash_list -= A
+			A:gib() //gib all non-carbon things like simple mobs
```
Comment (GinjaNinja32) [inline code/modules/tram/tram.dm]: likewise here and the other places just above

Comment (Vivalas) [inline code/modules/tram/tram.dm]: I could seperate it into handle_mob_crash and handle_obj_crash then

Comment (GinjaNinja32) [inline code/modules/tram/tram.dm]: Potentially, yes. Maybe make it `/atom/movable/proc/handle_tram_crash(obj/tram/tram_controller/tram, dir)` and just call `A.handle_tram_crash(src, dir)`

Comment (Zuhayr) [inline code/modules/tram/tram.dm]: Would be better to define an atom level tram_act() like that for consistency's sake.

**Comment by Vivalas** [comment]
This also auto-gibs cyborgs I think, not sure if that is a good thing, can fix.

**Comment by Vivalas2** [comment]
@Zuhayr D:

**Comment by Vivalas2** [comment]
Decided to implement the atom level tram_act

**Pull request merged**